ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -4721,11 +4727,14 @@ mod tests {
                 cmp r.32.x, 0x03
                 setz r.8._
                 ; guard true, %1, [] ; ...
-                bt r.32._, 0x00
+                and r.32.y, 0x01
+                mov [rbp-0x01], r.8.y
```

Comment:
The reason for this extra `mov` here is that we are now spilling to free up a register needed for the new deopt jump. I'm not super sure about the extra `and`. Is this because the value we are spilling is an `i8` and we need to make sure we are not writing any additional bytes?

Reply:
The `and` is because it's an `i1` with `Undefined` upper bits: it must be zero extended before spilling to the stack.